extensions/pip/main.ts

```diff
@@ -0,0 +1,214 @@
+import {
+  green,
+  red,
+  yellow,
+} from "https://deno.land/std@0.150.0/fmt/colors.ts";
+import { PhylumApi } from "https://deno.phylum.io/phylum.ts";
+
+// List with all of pip's subcommands.
+const knownSubcommands = [
+  "install",
+  "download",
+  "uninstall",
+  "freeze",
+  "inspect",
+  "list",
+  "show",
+  "check",
+  "config",
+  "search",
+  "cache",
+  "index",
+  "wheel",
+  "hash",
+  "completion",
+  "debug",
+  "help",
+];
+
+// Ensure the first argument is a known subcommand.
+//
+// This prevents us from skipping the analysis when an argument is passed before
+// the first subcommand (i.e.: `pip --no-color install package`).
+const subcommand = Deno.args[0];
+if (Deno.args.length != 0 && !knownSubcommands.includes(subcommand)) {
+  console.error(
+    `[${
+      red("phylum")
+    }] This extension does not support arguments before the first subcommand. Please open an issue if "${subcommand}" is not an argument.`,
+  );
+  Deno.exit(125);
+}
+
+// Ignore all commands that shouldn't be intercepted.
+if (Deno.args.length == 0 || subcommand != "install") {
+  const cmd = Deno.run({ cmd: ["pip3", ...Deno.args] });
+  const status = await cmd.status();
+  Deno.exit(status.code);
+}
+
+// Analyze new dependencies with phylum before install/update.
+await checkDryRun();
+
+// Perform the package installation.
+const installStatus = PhylumApi.runSandboxed({
```

Comment:
Was the intention to "perform the package installation" here _outside_ of the sandbox? A `Deno.run()` was expected here instead of another `PhylumApi.runSandboxed()`.

Reply:
The install is intentionally sandboxed. This is a feature of this extension.